Automattic/wp-calypso
# Replace overview when site is with a pending migration

Related to #95379 and #95380

## Proposed Changes

* Extract some components from the "Hosting Features" tab, so we can reuse it.
* It customizes the Sites overview with a button to continue the migration when it's pending.
* The button contains a `ref` as `hosting-migration-overview`, so we can identify when the user is coming from this page.

#### This PR Doesn't Include

* Disable other tabs from the menu.
* Update the header when navigating to the site, similar to the changes we have in the listing.

@donnapep, do you think it's fine to deploy it as it is and implement these other parts in other PRs? I think one issue would be fine for both changes. I'll wait for your answer to then create the new issue, otherwise, I'll continue working on this one.

## Why are these changes being made?


* We want to offer a way for the users to continue their migration if they stopped in the middle.

## Testing Instructions



* If https://github.com/Automattic/wp-calypso/pull/95413 wasn't merged yet, merge it into this branch to make tests more real and easier.
* Navigate to the flow `/setup/migration`.
* Navigate until the "How to migrate" step.
* Navigate to `/sites`.
* Click on the site, and click on "Start your migration".
* Make sure you navigate back to where you were in the flow.
* Click on the "Do it for me" option.
* Navigate again to the `/sites` and click on "Start your migration".
* Make sure you navigate back to your choice.
* Return one step and choose the "Do it myself".
* Navigate to the `/sites`.
* Check that you won't see the button because it changes quickly to the "started" status that will be implemented as part of https://github.com/Automattic/wp-calypso/issues/95383.
* Now repeat the tests using the flow `/setup/hosted-site-migration` and make sure it works properly.

## Screenshots

<img width="1399" alt="Screenshot 2024-10-18 at 18 40 00" src="https://github.com/user-attachments/assets/ec499f54-0781-46c9-b52d-19e7d5e5e741">

## Pre-merge Checklist
